### PR TITLE
[MO] Optimize redundant Concat in GRUBlockCell conversion

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - Optimize redundant Concat in GRUBlockCell conversion,
 Change `Concat->Split->Concat` subgraph to `Split->Concat` 
 (based on @mateusztabaka review https://github.com/openvinotoolkit/openvino/pull/11732#discussion_r883667743)

### Tickets:
 - 84821, (83334)

### CODE REVIEW

**Review by ceciliapeng2011 — APPROVED** [review]
LGTM

**Review by mbencer — APPROVED** [review]

**Review by itikhono — APPROVED** [review]

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py [hunk 1]
```diff
@@ -26,69 +26,52 @@ def find_and_replace_pattern(self, graph: Graph):
...
-            split_rzh_b.out_port(2).connect(concat_zrh_b.in_port(2))
+            # B_z -> z at 0
+            split_rz_b.out_port(1).connect(concat_zrh_b.in_port(0))
 
-            concat_b.out_port(0).connect(split_rzh_b.in_port(0))
+            # B_h -> h at 2
+            tf_gru_block_cell.in_port(5).get_connection().set_destination(concat_zrh_b.in_port(2))
 
             permute_order = int64_array([1, 0])
             transpose_w = create_op_node_with_second_input(graph, Transpose, permute_order,
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:77]: please preserve name from the original node and also preserve tensor names

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py [hunk 2]
```diff
@@ -26,69 +26,52 @@ def find_and_replace_pattern(self, graph: Graph):
             tf_gru_block_cell.in_port(0).get_connection().set_destination(new_gru_cell.in_port(0))
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:26]: please sort out imports and clean-up

Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:26]: please add description for this transformation about what it does

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py [hunk 3]
```diff
@@ -26,69 +26,52 @@ def find_and_replace_pattern(self, graph: Graph):
...
-            split_rzh_b.out_port(1)
-            concat_zrh_b = Concat(graph, {'name': original_name + '/Concat_zrh_B',
+            # B_rz -> B_r, B_z
+            tf_gru_block_cell.in_port(4).get_connection().set_destination(split_rz_b.in_port(0))
+
+            concat_zrh_b = Concat(graph, {'name': original_name + '/Concat_B_zrh',
                                                    'axis': 0}).create_node()
             concat_zrh_b.add_input_port(0)
             concat_zrh_b.add_input_port(1)
             concat_zrh_b.add_input_port(2)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:65]: please use `in_ports_count` attribute for `Concat` instead

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py [hunk 4]
```diff
@@ -26,69 +26,52 @@ def find_and_replace_pattern(self, graph: Graph):
...
-            concat_zrh_w = Concat(graph, {'name': original_name + '/Concat_zrh_W',
+            concat_zrh_w = Concat(graph, {'name': original_name + '/Concat_W_zrh',
                                                    'axis': 1}).create_node()
             concat_zrh_w.add_input_port(0)
             concat_zrh_w.add_input_port(1)
             concat_zrh_w.add_input_port(2)
 
-            # r at 0 -> r at 1
-            split_rzh_w.out_port(0).connect(concat_zrh_w.in_port(1))
+            # W_r -> r at 1
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:43]: please make more descriptive comments

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 2 above)
Comment (mitruska) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:30]: Sorted out (but I've not found nothing to clean up).

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 4 above)
Comment (mitruska) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py]: Done (I believe;)

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 3 above)
Comment (mitruska) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py]: Good point, done

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 2 above)
Comment (mitruska) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:30]: Done

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 1 above)
Comment (rkazants) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:73]: I think we are fine with tensor names. Just one comment above how we swap names between new node and old one to be replaced

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py [hunk 5]
```diff
@@ -23,76 +26,52 @@ def find_and_replace_pattern(self, graph: Graph):
             new_gru_cell = GRUCell(graph, {}).create_node()
             rename_node(new_gru_cell, original_name)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:27]: Please use:
```
rename_nodes([(tf_gru_block_cell, original_name + '/to_be_removed'), (new_gru_cell , original_name)])
```

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py [hunk 6]
```diff
@@ -23,76 +26,52 @@ def find_and_replace_pattern(self, graph: Graph):
...
+                                          'axis': 0}).create_node()
 
-            # h at 2 -> h at 2
-            split_rzh_b.out_port(2).connect(concat_zrh_b.in_port(2))
+            # Swap and concat gates: B_rz -> B_zr
+            split_rz_b.out_port(0).connect(concat_zrh_b.in_port(1))
+            split_rz_b.out_port(1).connect(concat_zrh_b.in_port(0))
 
-            concat_b.out_port(0).connect(split_rzh_b.in_port(0))
+            # Conncat B_h gate: B_zr -> B_zrh
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:68]: `Concat`

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py [hunk 7]
```diff
@@ -23,76 +26,52 @@ def find_and_replace_pattern(self, graph: Graph):
...
+            tf_gru_block_cell.in_port(5).get_connection().set_destination(concat_zrh_b.in_port(2))
 
+            # Transpose W Shape [input_size + hidden_size, 3 * hidden_size] to [3 * hidden_size, input_size + hidden_size]
             permute_order = int64_array([1, 0])
             transpose_w = create_op_node_with_second_input(graph, Transpose, permute_order,
-                                                          dict(name=original_name + 'Transpose_W'), concat_zrh_w)
+                                                          dict(name=original_name + '/Transpose_W'), concat_zrh_w)
 
             transpose_w.out_port(0).connect(new_gru_cell.in_port(2))
             concat_zrh_b.out_port(0).connect(new_gru_cell.in_port(3))
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:77]: Please use instead for removing nodes:
```
graph.remove_nodes_from([tf_gru_block_cell.id])
``

**Review by rkazants — COMMENTED** [review]
Just last two comments. The other looks good. Thanks!

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 5 above)
Comment (mitruska) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py]: Done

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 6 above)
Comment (mitruska) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py]: Done

File: tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py (hunk 7 above)
Comment (mitruska) [reply tools/mo/openvino/tools/mo/front/tf/GRUBlockCellReplacement.py:74]: Done

**Review by rkazants — APPROVED** [review]

**Pull request merged**